draft.tex

```diff
@@ -770,6 +775,33 @@ \section*{Outlook}
 attitude of the scientific community towards the publication,
 preservation, and verification of computer-aided research.
 
+% -> Reproducible software environments
+
+A fundamental technical issue is the difficulty of archiving or
+accurately describing the software environments in which computational
+scientists perform their work. A publication should be accompanied by
+both a human-readable description of this environment and an
+executable binary form. The human-readable description allows an
+inspection of the versions of all software packages that were used,
+for example to check for the impact of bugs that become known only
+after a study was published. The executable version enables other
+scientists to re-run the analyses and inspect intermediate
+results. Ideally, the human-readable description would permit
+rebuilding the executable version, in the same way that software
+source code permits rebuilding executable binaries. This approach is
+pursued for example by the package manager Guix \citep{Courtes:2015}.
+A more limited but still useful implementation of the same idea exists
+in the form of the conda package manager
+\citep{conda}. Containerization, as implemented e.g. by Docker
```

Comment:
For conda, add "that allows one to store a set of dependencies in a file ("environment.yml", see [Creating an environment from an environment.yml file](https://conda.io/docs/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file) and [Exporting the environment file](https://conda.io/docs/user-guide/tasks/manage-environments.html#exporting-the-environment-file)), thus communicating them explicitly.

Reply:
Good point. I have extended the discussion of conda a bit.